### PR TITLE
Fix warnings

### LINKED ISSUE #7: Compiler Warnings
When compiling with gcc -Wall:

```
Warnings   << swri_profiler:make logs/swri_profiler/build.make.000.log                                                                                                                                       
src/swri_profiler/swri_profiler/src/nodes/basic_profiler_example_node.cpp: In member function ‘int BasicProfilerExampleNode::superSlowFibonacci(int)’:
src/swri_profiler/swri_profiler/src/nodes/basic_profiler_example_node.cpp:63:3: warning: no return statement in function returning non-void [-Wreturn-type]
   }
   ^
cd build/swri_profiler; catkin build --get-env swri_profiler | catkin env -si  /usr/bin/make --jobserver-fds=6,7 -j; cd -
```

```
Warnings   << swri_profiler_tools:make logs/swri_profiler_tools/build.make.000.log                                                                                                                           
src/swri_profiler/swri_profiler_tools/src/profile.cpp:303:13: warning: ‘bool swri_profiler_tools::compareInitialStringList(const QStringList&, const QStringList&)’ defined but not used [-Wunused-function]
 static bool compareInitialStringList(
             ^
cd build/swri_profiler_tools; catkin build --get-env swri_profiler_tools | catkin env -si  /usr/bin/make --jobserver-fds=6,7 -j; cd 
```

### PR DESCRIPTION
**Pull request by evenator**
FIxes #7

### CODE REVIEW

**Comment by evenator** [comment]
@elliotjo I don't have merge rights here. Does this look good to you?

File: swri_profiler/src/nodes/basic_profiler_example_node.cpp [hunk 1]
```diff
@@ -56,7 +56,7 @@ class BasicProfilerExampleNode
     }
   }
 
-  int superSlowFibonacci(int x)
+  void superSlowFibonacci(int x)
```
Comment (elliotjo) [inline swri_profiler/src/nodes/basic_profiler_example_node.cpp:59]: This change is good.

File: swri_profiler_tools/src/profile.cpp [hunk 2]
```diff
@@ -300,7 +300,7 @@ void Profile::rebuildFlatIndex()
 }
 
 // Compares the first N items of two string lists.
-static bool compareInitialStringList(
+bool compareInitialStringList(
```
Comment (elliotjo) [inline swri_profiler_tools/src/profile.cpp:303]: I'm not a fan of this change.  Why do you want to remove it?  Labeling a function static makes it very clear that it cannot be not called from anywhere outside this file, which is great for knowing the risk of changing it.

Comment (evenator) [reply swri_profiler_tools/src/profile.cpp:303]: I did this so long ago that I don't actually remember, but I suspect it was related to this warning:

```
Warnings   << swri_profiler_tools:make logs/swri_profiler_tools/build.make.000.log                                                                                                                           
src/swri_profiler/swri_profiler_tools/src/profile.cpp:303:13: warning: ‘bool swri_profiler_tools::compareInitialStringList(const QStringList&, const QStringList&)’ defined but not used [-Wunused-function]
 static bool compareInitialStringList(
             ^
cd build/swri_profiler_tools; catkin build --get-env swri_profiler_tools | catkin env -si  /usr/bin/make --jobserver-fds=6,7 -j; cd 
```

It seems that the actual problem here is that the function is never used (in this file or elsewhere). Is this a half-implemented feature, or should it be removed?

**Pull request closed without merging**